index.js

```diff
@@ -1,27 +1,38 @@
-"use strict";
+'use strict';
 
 class CleanTerminalPlugin {
   constructor(options = {}) {
-    this.message = options.message;
+    const {message, onlyInWatchMode = true} = options;
+    this.message = message;
+    this.onlyInWatchMode = onlyInWatchMode;
   }
 
   apply(compiler) {
     if (compiler.hooks) {
-      compiler.hooks.afterCompile.tap("CleanTerminalPlugin", () => {
-        if (process.env.NODE_ENV !== "production") this.clearConsole();
+      compiler.hooks.afterCompile.tap('CleanTerminalPlugin', () => {
+        if (this.shouldClearConsole(compiler)) this.clearConsole();
       });
     } else {
       // backwards compatible version of compiler.hooks
-      compiler.plugin("emit", (_, done) => {
-        if (process.env.NODE_ENV !== "production") this.clearConsole();
+      compiler.plugin('emit', (_, done) => {
+        if (this.shouldClearConsole()) this.clearConsole();
```

Comment:
There's an error here (which is currently still in in the NPM package). 
The compiler is not passed to the shouldClearConsole method which triggers an undefined-error ;)

Reply:
Thanks for the heads up, created #12

Reply:
I added a PR that fixed the issue :)

Reply:
Thanks for the catch @mrnocreativity. Apologies guys, I should've spent more time looking this over before pr-ing 😞

Reply:
No worries :) That's what community is for!

Reply:
@andykais no worries, I also missed it while reviewing :)
@mrnocreativity Thanks for the PR, I wasn't sure you'd be sending one, so I created an issue as well.

Moving the conversation to #11